File: pset2/readabibity/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int count_letters(string s);
int count_words(string s);
int count_sentences(string s);
double colemanLiauIndex(int l, int w, int s);

int main(void)
{
    string s = get_string("Text: ");
    int l = count_letters(s);
    int w = count_words(s);
    int ss = count_sentences(s);
    
    printf("%i letter(s)\n", l);
    printf("%i word(s)\n", w);
    printf("%i sentence(s)\n", ss);

    int c = round(colemanLiauIndex(l, w, ss));
    if (c < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (c > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", c);
    }
}

int count_letters(string s) 
{
    int n = strlen(s);
    int letterNum = 0;
    for (int i = 0; i < n; i++)
    {
        if ((s[i] >= 65 && s[i] <= 90) || (s[i] >= 97 && s[i] <= 122)) 
        {
            letterNum++;
        }
    }
    
    return letterNum;
}

int count_words(string s)
{
    int n = strlen(s);
    int blankNum = 0;
    for (int i = 0; i < n; i++)
    {
        if (s[i] == 32) 
        {
            blankNum++;
        }
    }
    
    return blankNum + 1;
}

int count_sentences(string s)
{
    int n = strlen(s);
    int sentenceNum = 0;
    for (int i = 0; i < n; i++)
    {
        if (s[i] == 33 || s[i] == 46 || s[i] == 63) 
        {
            sentenceNum++;
        }
    }
    
    return sentenceNum;
}

double colemanLiauIndex(int l, int w, int s)
{
    return 0.0588 * ((double) l * 100 / (double) w) - 0.296 * ((double)s * 100 / (double)w) - 15.8;
}

File: pset6/readability/readability.py
from cs50 import get_string

text = get_string("Text: ")

lCount = 0
wCount = 1
sCount = 0

for letter in text:
    if (ord(letter) >= 65 and ord(letter) <= 90) or (ord(letter) >= 97 and ord(letter) <= 122):
        lCount += 1
    if ord(letter) == 32:
        wCount += 1
    if ord(letter) == 33 or ord(letter) == 46 or ord(letter) == 63:
        sCount += 1

# print(lCount)
# print(wCount)
# print(sCount)


def colemanLiauIndex(l, w, s):
    return 0.0588 * (l * 100 / w) - 0.296 * (s * 100 / w) - 15.8


index = round(colemanLiauIndex(lCount, wCount, sCount))
# print(index)


if index < 1:
    print("Before Grade 1")
elif index > 16:
    print("Grade 16+")
else:
    print("Grade %i" % index)